MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/javaservices/MultipleGeometriesDirectionsRouteActivity.java

```diff
@@ -0,0 +1,294 @@
+    if (loadedMapStyle.getLayer(SETTLEMENT_LABEL_LAYER_ID) != null) {
+      loadedMapStyle.addLayerBelow(directionsRouteLineLayer, SETTLEMENT_LABEL_LAYER_ID);
+    } else {
+      loadedMapStyle.addLayer(directionsRouteLineLayer);
+    }
+  }
+
+  /**
+   * Create and style a CircleLayer that will place circles for each of the Mapbox Directions API
+   * route's step maneuver locations.
+   * @param loadedMapStyle the map's {@link Style} object
+   */
+  private void initStepManeuverCircleLayer(@NonNull Style loadedMapStyle) {
+    CircleLayer individualCirclesLayer = new CircleLayer(STEPS_CIRCLE_LAYER_ID, CIRCLE_GEOJSON_SOURCE_ID);
+    individualCirclesLayer.setProperties(
+      circleColor(CIRCLE_COLOR),
+      circlePitchAlignment(ALIGN_CIRCLES_WITH_MAP ? CIRCLE_PITCH_ALIGNMENT_MAP : CIRCLE_PITCH_ALIGNMENT_VIEWPORT),
+      circleRadius(CIRCLE_RADIUS));
+    loadedMapStyle.addLayer(individualCirclesLayer);
+  }
+
+  /**
+   * Create and style a CircleLayer that will place circles beneath the STEPS_CIRCLE_LAYER_ID layer
+   * for each of the Mapbox Directions API route's step maneuver locations.
+   * @param loadedMapStyle the map's {@link Style} object
+   */
+  private void initStepManeuverBackgroundCircleLayer(@NonNull Style loadedMapStyle) {
+    CircleLayer individualCirclesLayer = new CircleLayer(STEPS_BACKGROUND_CIRCLE_LAYER_ID, CIRCLE_GEOJSON_SOURCE_ID);
+    individualCirclesLayer.setProperties(
+      circleColor(BACKGROUND_CIRCLE_COLOR),
+      circlePitchAlignment(ALIGN_CIRCLES_WITH_MAP ? CIRCLE_PITCH_ALIGNMENT_MAP : CIRCLE_PITCH_ALIGNMENT_VIEWPORT),
+      circleRadius(CIRCLE_RADIUS + RADIUS_DIFFERENCE_BETWEEN_WAYPOINT_CIRCLES_AND_BACKGROUND_CIRCLES));
+    loadedMapStyle.addLayerBelow(individualCirclesLayer, STEPS_CIRCLE_LAYER_ID);
+  }
+
+  /**
+   * Make a request to the Mapbox Directions API. Once successful, pass the route to the
+   * route layer.
+   *
+   * @param origin      the starting point of the route
+   * @param destination the desired finish point of the route
+   */
+  private void getRoute(Point origin, Point destination) {
+    MapboxDirections directionsApiClient = MapboxDirections.builder()
+      .origin(origin)
+      .destination(destination)
+      .overview(DirectionsCriteria.OVERVIEW_FULL)
+      .profile(DirectionsCriteria.PROFILE_DRIVING)
+      .steps(true)
+      .accessToken(getString(R.string.access_token))
+      .build();
+
+    directionsApiClient.enqueueCall(new Callback<DirectionsResponse>() {
+      @Override
+      public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
+        Timber.d("Response code: %s", response.code());
+        if (response.body() == null) {
+          Timber.e("No routes found, make sure you set the right user and access token.");
+          return;
+        } else if (response.body().routes().size() < 1) {
+          Timber.e("No routes found");
+          return;
+        }
+
+        if (response.body() != null) {
```

Comment:
This could be combined to the previous if statements, and we could remove the return for the first two cases.
```
        if (response.body() == null) {
          Timber.e("No routes found, make sure you set the right user and access token.");
        } else if (response.body().routes().size() < 1) {
          Timber.e("No routes found");
        } else {
           // Get the directions route
```

Reply:
Sure. Adjusted to your suggestion.